From: Matthew Leeds <matthew.leeds@endlessm.com>
Date: Tue, 21 Aug 2018 12:25:50 -0700
Subject: common/installation: Search dynamic remotes for appstream2 also

Flatpak has API called flatpak_installation_list_remotes_by_type() which
can list dynamic (LAN/USB) remotes that mirror configured remotes in an
installation. It does this by searching them for the appstream/<arch>
ref, such as appstream/x86_64. But Flatpak now supports
appstream2/<arch> as a way to provide the appstream data as uncompressed
XML, and it's possible that a USB created with `flatpak create-usb` (or
a LAN peer) only has the appstream2 ref available for a certain
collection ID. So this commit changes
list_remotes_for_configured_remote() so that it looks for both
appstream/<arch> and appstream2/<arch>, which makes
flatpak_installation_list_remotes_by_type() robust to that scenario.

(cherry picked from commit e9d9f54ab8fc7a6b66f56cea66e5a70b45e7e703)
---
 common/flatpak-installation.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/common/flatpak-installation.c b/common/flatpak-installation.c
index cf9a47d..fec8a5f 100644
--- a/common/flatpak-installation.c
+++ b/common/flatpak-installation.c
@@ -1115,8 +1115,10 @@ list_remotes_for_configured_remote (FlatpakInstallation *self,
 {
   g_autofree gchar *collection_id = NULL;
   OstreeCollectionRef ref;
-  const OstreeCollectionRef *refs[2] = { NULL, };
+  OstreeCollectionRef ref2;
+  const OstreeCollectionRef *refs[3] = { NULL, };
   g_autofree gchar *appstream_ref = NULL;
+  g_autofree gchar *appstream2_ref = NULL;
 
   g_auto(OstreeRepoFinderResultv) results = NULL;
   g_autoptr(GAsyncResult) result = NULL;
@@ -1144,6 +1146,10 @@ list_remotes_for_configured_remote (FlatpakInstallation *self,
   ref.collection_id = collection_id;
   ref.ref_name = appstream_ref;
   refs[0] = &ref;
+  appstream2_ref = g_strdup_printf ("appstream2/%s", flatpak_get_arch ());
+  ref2.collection_id = collection_id;
+  ref2.ref_name = appstream2_ref;
+  refs[1] = &ref2;
 
   if (types_filter[FLATPAK_REMOTE_TYPE_USB])
     {
